### PR TITLE
Frontend: Adding support to select preferred timezone for presentation of date and time values

### LINKED ISSUE #558: Support custom timezone selection
Is it possible to configure Grafana to always show only server time? I.e. if event has appeared in 00:00 on server, I want to see it only in 00:00 in Grafana independently on my browsers local timezone, which may be changed.

### PR DESCRIPTION
**Pull request by mckn**
**What this PR does / why we need it**:
There are multiple reasons why this is a handy feature but the following scenario explains the need pretty well. If you are monitoring a platform that operates in multiple time zones it would make everything easier to display the date/time values in the local timezone instead of your local or UTC.

Would help out with debugging, monitoring etc.

**Todo:**
- [x] Add possibility to select a custom time zone.
- [x] Rewrite TimePickerSettings to a react component.
- [x] Create helper functions for formatting date/time values.
- [x] Make sure we convert input date/time to utc date/time values.
- [x] Make sure we format date/time values with the new functionality when being displayed.
- [x] Add some kind of visual indication that times are displayed in special timeZones when it is overriden for a dashboard.
- [x] fix tooltip in react graph panel (explore only). The reason is this: https://github.com/grafana/grafana/blob/master/public/app/features/explore/state/reducers.ts#L534
- [x] fix failing test due to local/utc generation.

**Fixes:**
#558

**Special notes for your reviewer**:
The plan is to create an internal abstracting for formatting date/time values. It will use moment-timezone under the hood but do it in a way so we easily can replace moment in the future.

The code can be improved in several ways but I think this is a good enough implementation that can be shipped along the 7.0-beta1. Issue #23917 contains a list of things that will be improved.

### CODE REVIEW

**Review by torkelo — COMMENTED** [review]
Looks like a good start!

**Comment by ryantxu** [comment]
Make sure that the existing `valueFormatters` also respect the time zone.  If you search for `timeZone === 'utc'` you will find everything you need:

image

**Comment by mckn** [comment]
> Make sure that the existing `valueFormatters` also respect the time zone. If you search for `timeZone === 'utc'` you will find everything you need:
> 
> image

yes, that is the plan :)

File: packages/grafana-ui/src/components/TimePicker/TimeRangePicker.tsx [hunk 1]
```diff
@@ -226,7 +233,7 @@ const TimePickerButtonLabel = memo<Props>(props => {
   return (
     <span className={styles.container}>
       <span>{formattedRange(props.value, isUTC)}</span>
-      {isUTC && <span className={styles.utc}>UTC</span>}
+      <span className={styles.utc}>{dateTimeFormatZoneAbbrevation(Date.now(), { timeZone: props.timeZone })}</span>
```
Comment (mckn) [inline packages/grafana-ui/src/components/TimePicker/TimeRangePicker.tsx:236]: nit: will talk with @torkelo on how to resolve this

**Comment by torkelo** [comment]
If go to preferences and try to change the timezone we are not using the TimeZonePicker component (which uses the Cascader under the hood). Was there a reason for this?

**Review by torkelo — APPROVED** [review]
Took this for a quick spin and works as expected! amazing job! 

And the code looks really good.

But would be nice to use our TimeZonePicker, but TimeZonePicker is currently missing a Default and UTC option.

**Comment by mckn** [comment]
> Took this for a quick spin and works as expected! amazing job!
> 
> And the code looks really good.
> 
> But would be nice to use our TimeZonePicker, but TimeZonePicker is currently missing a Default and UTC option.

Yes, I will adjust that one in the next PR. I will show you how the behavior is when running that one. Think I need some feedback on how to make that one better.

**Comment by terrillmoore** [comment]
Thank you for this work!

**Comment by harvey-peakpower** [comment]
Thanks for your great job. 
Will these timezone selection work for Alert function? For example email alert ,Notification channels, and Graph image renderer using selected timezone.

**Comment by mckn** [comment]
> Thanks for your great job.
> Will these timezone selection work for Alert function? For example email alert ,Notification channels, and Graph image renderer using selected timezone.

Hopefully :)

**Comment by harvey-peakpower** [comment]
> > Thanks for your great job.
> > Will these timezone selection work for Alert function? For example email alert ,Notification channels, and Graph image renderer using selected timezone.
> 
> Hopefully :)

Thanks. Could you tell me when this feature can be released or published? My current project urgently needs this feature.

**Comment by mckn** [comment]
> > > Thanks for your great job.
> > > Will these timezone selection work for Alert function? For example email alert ,Notification channels, and Graph image renderer using selected timezone.
> > 
> > 
> > Hopefully :)
> 
> Thanks. Could you tell me when this feature can be released or published? My current project urgently needs this feature.

It should be included in the 7.x releases.

**Pull request merged**